### PR TITLE
IDENTITY-6651-DCR: Improve DCRMUtils unit test coverage.

### PR DESCRIPTION
**Pull request by hasinidilanka**

### CODE REVIEW

**Comment by hasinidilanka** [comment]
@nilasini

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java [hunk 1]
```diff
@@ -46,49 +46,34 @@ public void testIsRedirectionUriValid(String url, boolean response) throws Excep
...
-    @Test(dataProvider = "BuildServerException")
+    @Test(dataProvider = "BuildServerException", expectedExceptions = DCRMServerException.class)
     public void testGenerateServerException1(DCRMConstants.ErrorMessages error, String data) throws Exception {
-        try {
-            DCRMUtils.generateServerException(error,data);
-        } catch (DCRMServerException e){
-            Assert.assertEquals(e.getMessage(),data);
-        }
-
+        throw DCRMUtils.generateServerException(error,data);
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: Formatting issue.

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java [hunk 2]
```diff
@@ -46,49 +46,34 @@ public void testIsRedirectionUriValid(String url, boolean response) throws Excep
...
+
+    @Test(dataProvider = "BuildServerException", expectedExceptions = DCRMClientException.class)
     public void testGenerateClientException(DCRMConstants.ErrorMessages error, String data) throws Exception {
         Throwable e = new Throwable();
-        try {
-            DCRMUtils.generateClientException(error,data,e);
-        } catch (DCRMClientException ex) {
-            Assert.assertEquals(ex.getMessage(),data);
-        }
+        throw DCRMUtils.generateClientException(error,data,e);
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: Formatting issue.

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java [hunk 3]
```diff
@@ -46,49 +46,34 @@ public void testIsRedirectionUriValid(String url, boolean response) throws Excep
...
 
-    @Test(dataProvider = "BuildServerException")
+    @Test(dataProvider = "BuildServerException", expectedExceptions = DCRMClientException.class)
     public void testGenerateClientException1(DCRMConstants.ErrorMessages error, String data) throws Exception {
-        try {
-            DCRMUtils.generateClientException(error,data);
-        } catch (DCRMClientException e){
-            Assert.assertEquals(e.getMessage(),data);
-        }
+        throw DCRMUtils.generateClientException(error,data);
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: Formatting issue.

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java [hunk 4]
```diff
@@ -46,49 +46,34 @@ public void testIsRedirectionUriValid(String url, boolean response) throws Excep
...
+    @Test(dataProvider = "BuildServerException", expectedExceptions = DCRMServerException.class)
     public void testGenerateServerException(DCRMConstants.ErrorMessages error, String data) throws Exception {
         Throwable e = new Throwable();
-        try {
-            DCRMUtils.generateServerException(error,data,e);
-        } catch (DCRMServerException ex) {
-            Assert.assertEquals(ex.getMessage(), data);
-        }
-
+        throw DCRMUtils.generateServerException(error,data,e);
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: Formatting issue.

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java [hunk 5]
```diff
@@ -46,49 +46,34 @@ public void testIsRedirectionUriValid(String url, boolean response) throws Excep
...
     }
 
-    @Test(dataProvider = "BuildServerException")
+    @Test(dataProvider = "BuildServerException", expectedExceptions = DCRMServerException.class)
     public void testGenerateServerException1(DCRMConstants.ErrorMessages error, String data) throws Exception {
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: Can we use a more meaningful name than `testGenerateServerException1`

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java [hunk 6]
```diff
@@ -46,49 +46,34 @@ public void testIsRedirectionUriValid(String url, boolean response) throws Excep
...
+    @Test(dataProvider = "BuildServerException", expectedExceptions = DCRMClientException.class)
+    public void testThrowableClientException(DCRMConstants.ErrorMessages error, String data) throws Exception {
         Throwable e = new Throwable();
-        try {
-            DCRMUtils.generateClientException(error,data,e);
-        } catch (DCRMClientException ex) {
-            Assert.assertEquals(ex.getMessage(),data);
-        }
+        throw DCRMUtils.generateClientException(error, data, e);
+
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: remove the new line

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java [hunk 7]
```diff
@@ -46,49 +46,34 @@ public void testIsRedirectionUriValid(String url, boolean response) throws Excep
...
-    @Test(dataProvider = "BuildServerException")
-    public void testGenerateClientException1(DCRMConstants.ErrorMessages error, String data) throws Exception {
-        try {
-            DCRMUtils.generateClientException(error,data);
-        } catch (DCRMClientException e){
-            Assert.assertEquals(e.getMessage(),data);
-        }
+    @Test(dataProvider = "BuildServerException", expectedExceptions = DCRMClientException.class)
+    public void testGenerateClientException(DCRMConstants.ErrorMessages error, String data) throws Exception {
+        throw DCRMUtils.generateClientException(error, data);
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: remove extra new line

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java [hunk 8]
```diff
@@ -46,49 +46,34 @@ public void testIsRedirectionUriValid(String url, boolean response) throws Excep
...
-                {DCRMConstants.ErrorMessages.BAD_REQUEST_INVALID_INPUT,"error from bad request"}
+                {DCRMConstants.ErrorMessages.BAD_REQUEST_INVALID_INPUT, ""},
+                {DCRMConstants.ErrorMessages.BAD_REQUEST_INVALID_INPUT, "error from bad request"}
         };
     }
 
-    @Test(dataProvider = "BuildServerException")
-    public void testGenerateServerException(DCRMConstants.ErrorMessages error, String data) throws Exception {
+    @Test(dataProvider = "BuildServerException", expectedExceptions = DCRMServerException.class)
+    public void testThrowableServerException(DCRMConstants.ErrorMessages error, String data) throws Exception {
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: Add new line here

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java [hunk 9]
```diff
@@ -46,49 +46,34 @@ public void testIsRedirectionUriValid(String url, boolean response) throws Excep
...
-        try {
-            DCRMUtils.generateServerException(error,data,e);
-        } catch (DCRMServerException ex) {
-            Assert.assertEquals(ex.getMessage(), data);
-        }
+        throw DCRMUtils.generateServerException(error, data, e);
+    }
 
+    @Test(dataProvider = "BuildServerException", expectedExceptions = DCRMServerException.class)
+    public void testGenerateServerException(DCRMConstants.ErrorMessages error, String data) throws Exception {
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: Add new line

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java [hunk 10]
```diff
@@ -46,49 +46,34 @@ public void testIsRedirectionUriValid(String url, boolean response) throws Excep
...
-        } catch (DCRMServerException e){
-            Assert.assertEquals(e.getMessage(),data);
-        }
 
-    }
 
-    @Test(dataProvider = "BuildServerException")
-    public void testGenerateClientException(DCRMConstants.ErrorMessages error, String data) throws Exception {
+    @Test(dataProvider = "BuildServerException", expectedExceptions = DCRMClientException.class)
+    public void testThrowableClientException(DCRMConstants.ErrorMessages error, String data) throws Exception {
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: add new line

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java (hunk 8 above)
Comment (hasinidilanka) [reply components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: Fixed - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1242

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java (hunk 9 above)
Comment (hasinidilanka) [reply components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: Fixed - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1242

File: components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java (hunk 10 above)
Comment (hasinidilanka) [reply components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtilsTest.java]: Fixed - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1242

**Pull request merged**